### PR TITLE
fixes: #{}

### PR DESCRIPTION
**Pull request by itsmohitmkk**
Fixes : [#795](https://github.com/TheAlgorithms/Javascript/issues/795)
Implementation of Fenwick Tree.
Link for Description : https://www.geeksforgeeks.org/binary-indexed-tree-or-fenwick-tree-2/

### CODE REVIEW

**Comment by raklaptudirm** [comment]
Please read the contribution guidelines and change your code accordingly.

**Comment by itsmohitmkk** [comment]
Read the contribution guidelines. Sorry, I didn't follow it, I am new to open source. I made a few corrections to the code, please check if it's okay now.

File: Trees/FenwickTreeImplementation.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  index = index + 1;
+  while (index > 0) {
+    currSum += FeneickTree[index];
+    index -= index & (-index);
+  }
+    
+  return currSum;
+}
+
+let MAX_VALUE = 1000;
```
Comment (raklaptudirm) [inline Trees/FenwickTreeImplementation.js:35]: No global variables are allowed.

File: Trees/FenwickTreeImplementation.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+let FeneickTree = new Array(MAX_VALUE);
+let array = [3, 2, 0, 6, 5, -1, 2];
+
+let length = array.length;
+constructFenwickTree(array, length);
+
+getPrefixSum(6);
```
Comment (raklaptudirm) [inline Trees/FenwickTreeImplementation.js:42]: Testing is done in a separate `jest` file

File: Trees/FenwickTreeImplementation.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+array[2] += 6;
+updateFenwickTree(length, 2, 6);
+getPrefixSum(6);
+
+//Output1 : 17
+//Output2 : 23
```
Comment (raklaptudirm) [inline Trees/FenwickTreeImplementation.js:48]: The implemented functions should be exported.

**Review by raklaptudirm — CHANGES_REQUESTED** [review]
Format your code with `standard.js`.

File: Trees/FenwickTreeImplementation.js (hunk 3 above)
Comment (itsmohitmkk) [reply Trees/FenwickTreeImplementation.js]: `  return currSum
  }
}
export { FenwickTree }`
Exported the code

File: Trees/FenwickTreeImplementation.js (hunk 2 above)
Comment (itsmohitmkk) [reply Trees/FenwickTreeImplementation.js]: 1. Added file FenwickTreeImplementation.test.js for testing purposes where the functionality of Fenwick Tree is tested.
2. Removed all the global variables from FenwickTreeImplementation.js 
3. Formatted all the code with standard.js
4. Tested code using jest

File: Trees/FenwickTreeImplementation.js (hunk 1 above)
Comment (itsmohitmkk) [reply Trees/FenwickTreeImplementation.js]: Removed all the global variables from the file. Enclosed everything in class FenwicTree

File: Trees/FenwickTreeImplementation.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
+/*
+ * Author: Mohit Kumar
+ * Fedwick Tree Implementation in JavaScript
+ * Fedwick Tree Implementation for finding prefix sum.
+*/
+
+class FenwickTree {
+  constructFenwickTree (feneickArray, array, n) {
```
Comment (raklaptudirm) [inline Trees/FenwickTreeImplementation.js:8]: Rename this function to `constructor` so that is can be called during initialization,

**Review by raklaptudirm — CHANGES_REQUESTED** [review]

File: Trees/FenwickTreeImplementation.js (hunk 4 above)
Comment (itsmohitmkk) [reply Trees/FenwickTreeImplementation.js]: Renamed constructFenwickTree as a constructor.
Made required changes in FenwickTreeImplementation.test.js

**Review by raklaptudirm — CHANGES_REQUESTED** [review]
Remove "implementation" from the file names.

**Comment by itsmohitmkk** [comment]
> Remove "implementation" from the file names.

Changed name of files
Changed imports.

File: Trees/FenwickTree.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  constructor (feneickArray, array, n) {
+    for (let i = 1; i <= n; i++) {
+      feneickArray[i] = 0
+    }
+    for (let i = 0; i < n; i++) {
+      this.updateFenwickTree(feneickArray, n, i, array[i])
+    }
+  }
+
+  updateFenwickTree (feneickArray, n, index, value) {
```
Comment (raklaptudirm) [inline Trees/FenwickTree.js:17]:
```suggestion
  update (feneickArray, n, index, value) {
```
`FenwickTree` is in the class name, so no need.

**Review by raklaptudirm — CHANGES_REQUESTED** [review]

File: Trees/FenwickTree.js (hunk 5 above)
Comment (itsmohitmkk) [reply Trees/FenwickTree.js:17]: Changed `updateFenwickTree` to `update`

**Review by raklaptudirm — APPROVED** [review]

**Pull request merged**